src/test/java/io/openliberty/tools/common/config/ServerConfigDocumentOverridesTest.java

```diff
@@ -0,0 +1,246 @@
+    
+    // 1. variable default values in server.xml file
+    // 6. variable values declared in the server.xml file
+    @Test
+    public void processServerXml() throws FileNotFoundException, IOException, XPathExpressionException, SAXException {
+        File serversResourceDir = SERVERS_RESOURCES_DIR.toFile();
+        Document doc;
+        Map<String, File> libertyDirPropMap = new HashMap<String, File>();
+        libertyDirPropMap.put(ServerFeatureUtil.SERVER_CONFIG_DIR, serversResourceDir);
+
+        // no variables defined
+        ServerConfigDocument configDocument = new ServerConfigDocument(new TestLogger(), libertyDirPropMap);
+        File empty = new File(serversResourceDir, "emptyList.xml");
+        doc = configDocument.parseDocument(empty);
+        configDocument.parseVariablesForBothValues(doc);
+        assertTrue(configDocument.getDefaultProperties().isEmpty() && configDocument.getProperties().isEmpty());
+
+        // variables defined
+        configDocument = new ServerConfigDocument(new TestLogger(), libertyDirPropMap);
+        File defined = new File(serversResourceDir, "definedVariables.xml");
+        doc = configDocument.parseDocument(defined);
+        configDocument.parseVariablesForBothValues(doc);
+        assertEquals(1, configDocument.getDefaultProperties().size());
+        assertEquals("9080", configDocument.getDefaultProperties().getProperty("default.http.port"));
+        assertEquals(1, configDocument.getProperties().size());
+        assertEquals("9081", configDocument.getProperties().getProperty("http.port"));
+
+        // variables defined in <include/> files
+        configDocument = new ServerConfigDocument(new TestLogger(), libertyDirPropMap);
+        File include = new File(serversResourceDir, "testIncludeParseVariables.xml");
+        doc = configDocument.parseDocument(include);
+        assertTrue(configDocument.getDefaultProperties().isEmpty() && configDocument.getProperties().isEmpty());
+        configDocument.parseIncludeVariables(doc);
+        assertEquals(1, configDocument.getDefaultProperties().size());
+        assertEquals("9080", configDocument.getDefaultProperties().getProperty("default.http.port"));
+        assertEquals(1, configDocument.getProperties().size());
+        assertEquals("9081", configDocument.getProperties().getProperty("http.port"));
+        
+        // server.xml configDropins precedence
+        File serverConfigDir = SERVER_CONFIG_DIR.toFile();
+        libertyDirPropMap.put(ServerFeatureUtil.SERVER_CONFIG_DIR, serverConfigDir);
+        configDocument = new ServerConfigDocument(new TestLogger(), libertyDirPropMap);
+        doc = configDocument.parseDocument(new File(serverConfigDir, "server.xml"));
+        configDocument.processServerXml(doc);   // Variable resolution warnings can be ignored here
+        assertEquals("1", configDocument.getProperties().getProperty("config.dropins.defaults"));
```

Comment:
How am I supposed to know if these are correct? It would be nice to have a comment here explaining where each one is defined and overridden so I can tell if what I think should be the value is the actual value.